modules/JsonExporter/src/main/java/uk/ac/ox/oii/jsonexporter/JSONExporter.java

```diff
@@ -0,0 +1,277 @@
+                for (Node n : nodeArray) {
+
+                    String id = n.getId().toString();
+                    String label = n.getLabel();
+                    float x = n.x();
+                    float y = n.y();
+                    System.out.println(n.z());
+                    float size = n.size();
+                    String color = "rgb(" + (int) (n.r() * 255) + "," + (int) (n.g() * 255) + "," + (int) (n.b() * 255) + ")";
+
+                    /*if (renumber) {
+                     String newId=String.valueOf(nodeId);
+                     nodeIdMap.put(id,newId); 
+                     id=newId;
+                     nodeId++;
+                     }*/
+
+                    GraphNode jNode = new GraphNode(id);
+                    jNode.setLabel(label);
+                    jNode.setX(x);
+                    jNode.setY(y);
+                    jNode.setSize(size);
+                    jNode.setColor(color);
+
+                    
+                    for (Column col : attModel) {
+                            String name = col.getTitle();
+                            if (name.equalsIgnoreCase("Id") || name.equalsIgnoreCase("Label")
+                                    || name.equalsIgnoreCase("uid")) {
+                                continue;
+                            }
+
+                            Object valObj = n.getAttribute(col);
+                            if (valObj == null) {
+                                continue;
+                            }
+                            String val = valObj.toString();
+                            jNode.putAttribute(name, val);
+
+                    }
+
+                    jNodes.add(jNode);
+
+                    if (cancel) {
+                        return false;
+                    }
+                    Progress.progress(progress);
+                }
+
+
+                //Export edges. Progress is incremented at each step.
+                HashSet<GraphElement> jEdges = new HashSet<GraphElement>();
+                Edge[] edgeArray = graph.getEdges().toArray();
+
+                for (Edge e : edgeArray) {
+                    String sourceId = e.getSource().getId().toString();
+                    String targetId = e.getTarget().getId().toString();
+
+                    /*if (renumber) {
+                     sourceId = nodeIdMap.get(sourceId);
+                     targetId = nodeIdMap.get(targetId);
+                     }*/
+
+
+                    GraphEdge jEdge = new GraphEdge(String.valueOf(e.getId()));
+                    jEdge.setSource(sourceId);
+                    jEdge.setTarget(targetId);
+                    jEdge.setSize(e.getWeight());
+
+                    boolean mixColors=false;
+                    String color="";
+                    jEdge.setLabel(e.getLabel());
+
+                    float r=e.r();
+                    float g=e.g();
+                    float b=e.b();
+
+                    if (r==-1 || g==-1 || b==-1) {
```

Comment:
The way we mark edges NOT having a color has changed and you should rather test whether alpha == 0